### PR TITLE
Add /help formspec for commands and privileges

### LINKED ISSUE #7095: /help and /mods with many mods
##### Issue type

 - Feature request
 
##### Minetest version
```
Minetest 0.5.0-dev (Linux)
Using Irrlicht 1.8.4
BUILD_TYPE=None
RUN_IN_PLACE=0
USE_GETTEXT=1
USE_SOUND=1
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=1
STATIC_SHAREDIR="/usr/share/minetest"
```

##### OS / Hardware

Operating system: Lubuntu 17.10
CPU: intel

##### Summary

https://github.com/minetest/minetest/issues/7078#issuecomment-368623497
When many mods are installed, /help and /mods output is difficult to read. I think this should be fixed.
screenshot_20180303_220723
screenshot_20180303_220733

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes #7095

Chat commands
Privileges

This is the first implementation of a graphical back-end for the chat commands and privileges. Yet it is only for SSM, but could be extended to show the CSM chat commands as well.
There are yet no translations because the game part of builtin does not support that yet.

**What's changed:**
1) `/help` shows up the chat command list
2) `/help privs` shows up the privilege list
`/help <anything else>` is unaltered and will behave the same way as before.

### CODE REVIEW

**Comment by DS-Minetest** [comment]
:-1: 
The information from the formspec does not consist after closing it. This is very inconvenient. I prefer having all the information available in chat. (Eg. there might be a chatcommand called "haskdhfbl", I would not be able to remember how to write it exactly and would have to use help many times.)

**Comment by archfan7411** [comment]
>there might be a chatcommand called "haskdhfbl", I would not be able to remember how to write it exactly and would have to use help many times.

Perhaps it could put the details of a command in chat if the command in the list is clicked?

**Comment by benrob0329** [comment]
Copy+Paste, or command completion would fix that.

**Comment by SmallJoker** [comment]
@archfan7411 Good idea. I will add that.
@benrob0329 To have a smooth autocompletion, a CSM counterpart is needed (#5532). Chat inputs are only sent on enter, so for each autocompletion you'd currently have to wait for the server - which is what I want to avoid with this static formspec.

**Comment by SmallJoker** [comment]
The most recent commit adds a description textarea plus copying the command to the chat history by double-clicking an entry. Printing the description to the chat is not that helpful because the formspec shadow effect may make the text hard to read.
Is that satisfying, or do you have any suggestions to improve it further?

File: builtin/common/information_formspecs.lua [hunk 1]
```diff
@@ -40,24 +50,34 @@ end
 
 core.after(0, load_mod_command_tree)
 
-local function build_chatcommands_formspec(name)
+local function build_chatcommands_formspec(name, sel, copy)
 	local rows = {}
 	rows[1] = "#FFF,0,Command,Parameters"
 
+	local description = "For more information, click on any entry in the list.\n" ..
+		"Double-click to copy the entry to the chat history."
```
Comment (ClobberXD) [inline builtin/common/information_formspecs.lua]: `Double-click to copy the command to chat` sounds better IMHO. Also, the word `history` is sure to confuse many.

**Comment by ClobberXD** [comment]
Also, would the copying to chat feature work on touchscreens? i.e. Is there any way to "double-click" table entries? I'm guessing it's just double-tap, but just wanted to confirm.

**Comment by SmallJoker** [comment]
@ClobberXD Yes, that would be touble-tab.

**Comment by ClobberXD** [comment]
Ok, good to know. Testing... :)

**Comment by ClobberXD** [comment]
Works as expected. :+1: 

However, is it possible to clear the description area when the list of commands has been collapsed?

**Comment by SmallJoker** [comment]
@ClobberXD That's not possible because the client does not send the formspec fields for this event.

**Comment by ClobberXD** [comment]
Ah ok. It would be nice if that was possible. :)

**Comment by Enrikoo** [comment]
Nice Suggestion for /help command, works very smoothly for me in the game :)

**Comment by LoneWolfHT** [comment]
This looks great! Can't wait for it to be merged

**Pull request merged**